storage/innobase/fil/fil0crypt.cc

```diff
@@ -2993,3 +2994,172 @@ bool fil_space_verify_crypt_checksum(byte *page, ulint page_size,
 
   return (encrypted);
 }
+
+redo_log_key *redo_log_keys::load_latest_key(bool try_generate) {
+  size_t klen = 0;
+  char *key_type = nullptr;
+  byte *rkey = nullptr;
+
+  if (my_key_fetch(PERCONA_REDO_KEY_NAME, &key_type, nullptr,
+                   reinterpret_cast<void **>(&rkey), &klen) ||
+      rkey == nullptr || strncmp(key_type, "AES", 4) != 0) {
+    /* There is no key yet, we'll try to generate one */
+    my_free(rkey);
+    if (!try_generate) {
```

Comment:
change this to return(try_generate ? generate_and_store_new : nullptr);

Reply:
I'm not sure if that would make it more readable - in my experience hiding things like this in the ?: operator just makes things worse.

Reply:
true for some cases. here it looks clean

generate ? call_generate_function : nullptr